### PR TITLE
Add BUSL license text & bump node version

### PR DESCRIPTION
**Pull request by QGarchery**

### CODE REVIEW

**Review by MerlinEgalite — APPROVED** [review]

**Comment by QGarchery** [comment]
Putting as draft because the dual licensing seems to be on purpose. Maybe we should append the full BUSL license as well

File: LICENSE [hunk 1]
```diff
@@ -281,4 +281,109 @@ YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER
...
+
+Parameters
+
+Licensor:             Morpho Association
+
+Licensed Work:        Morpho Blue Core
+                      The Licensed Work is (c) 2023 Morpho Association
+
+Additional Use Grant: Any uses listed and defined at
+                      morpho-blue-core-license-grants.morpho.eth
```
Comment (MathisGD) [inline LICENSE:302]: this i guess must be updated

Comment (QGarchery) [reply LICENSE:302]: Ah yes. Not sure how to do it, especially the "change date" which should be void I think. I guess it should be explained that the users can choose the license as well

**Review by Rubilmax — APPROVED** [review]

**Comment by MathisGD** [comment]
Pending discussion on the dual licence

**Comment by QGarchery** [comment]
Closing this, let's reopen when needed

**Pull request closed without merging**